### PR TITLE
fix shellcheck in kube-aggregator

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

fix shellcheck failures in `staging/src/k8s.io/kube-aggregator`

**Which issue(s) this PR fixes**:

work towards #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/kube-aggregator/hack/register-all-apis-from.sh [hunk 1]
```diff
@@ -53,38 +53,40 @@ AGG_KUBECONFIG=${4}
...
 	resourceName=${version}.${group}
-	resourceFileName=${dir}/${resourceName}.yaml
-	cp ${scriptDir}/apiservice-template.yaml ${resourceFileName}
-	${SED} -i "s/RESOURCE_NAME/${resourceName}/" ${resourceFileName}
-	${SED} -i "s/API_GROUP/${group}/" ${resourceFileName}
-	${SED} -i "s/API_VERSION/${version}/" ${resourceFileName}
-	${SED} -i "s/SERVICE_NAMESPACE/${SERVICE_NAMESPACE}/" ${resourceFileName}
-	${SED} -i "s/SERVICE_NAME/${SERVICE_NAME}/" ${resourceFileName}
-	${SED} -i "s/CA_BUNDLE/${caBundle}/" ${resourceFileName}
+	resourceFileName=${dir}/"${resourceName}".yaml
```
Comment (BenTheElder) [inline staging/src/k8s.io/kube-aggregator/hack/register-all-apis-from.sh]: let's quote the whole string on these, like:
```suggestion
	resourceFileName="${dir}/${resourceName}.yaml"
```

**Comment by danielqsj** [comment]
@BenTheElder thanks, fixed the quota style, PTAL

File: staging/src/k8s.io/kube-aggregator/hack/build-image.sh [hunk 2]
```diff
@@ -24,7 +24,7 @@ function cleanup {
 }
 trap cleanup EXIT
 
-pushd "${KUBE_ROOT}/vendor/k8s.io/kube-aggregator"
+pushd "${KUBE_ROOT}/vendor/k8s.io/kube-aggregator" || exit
```
Comment (sttts) [inline staging/src/k8s.io/kube-aggregator/hack/build-image.sh]: this must be `exit 1`

File: staging/src/k8s.io/kube-aggregator/hack/register-all-apis-from.sh [hunk 3]
```diff
@@ -53,38 +53,40 @@ AGG_KUBECONFIG=${4}
...
+	${SED} -i "s/SERVICE_NAMESPACE/${SERVICE_NAMESPACE}/" "${resourceFileName}"
+	${SED} -i "s/SERVICE_NAME/${SERVICE_NAME}/" "${resourceFileName}"
+	${SED} -i "s/CA_BUNDLE/${caBundle}/" "${resourceFileName}"
 	echo "registering ${resourceName} using ${resourceFileName}"
 
-	kubectl --kubeconfig=${AGG_KUBECONFIG} create -f ${resourceFileName}
+	kubectl --kubeconfig="${AGG_KUBECONFIG}" create -f "${resourceFileName}"
 fi
 
-groupVersions=( $(kubectl --kubeconfig=${FROM_KUBECONFIG} get --raw / | grep /apis/ | sed 's/",.*//' | sed 's|.*"/apis/||' | grep '/') )
```
Comment (sttts) [inline staging/src/k8s.io/kube-aggregator/hack/register-all-apis-from.sh]: why this change?

Comment (danielqsj) [reply staging/src/k8s.io/kube-aggregator/hack/register-all-apis-from.sh]: this line hit this https://www.shellcheck.net/wiki/SC2207 .

File: staging/src/k8s.io/kube-aggregator/hack/build-image.sh (hunk 2 above)
Comment (danielqsj) [reply staging/src/k8s.io/kube-aggregator/hack/build-image.sh]: fixed, thanks :)

**Comment by spiffxp** [comment]
/retest
I think kops is failing due to https://github.com/kubernetes/kubernetes/issues/73444

**Comment by danielqsj** [comment]
@deads2k can you help review it? thanks

**Comment by fejta** [comment]
/assign @deads2k @lavalamp @smarterclayton
please /approve

**Comment by danielqsj** [comment]
PR rebased.

@deads2k @lavalamp @smarterclayton ping

**Comment by fejta** [comment]
@deads2k @lavalamp @smarterclayton
We still need approval for kube-aggregator
